pkg/yamlmeta/node.go

```diff
@@ -48,32 +49,41 @@ func (ds *DocumentSet) SetValue(val interface{}) error {
 }
 
 func (d *Document) SetValue(val interface{}) error {
-	d.Value = val
-	return nil
+	d.ResetValue()
+	return d.AddValue(val)
 }
 
 func (m *Map) SetValue(val interface{}) error {
 	return fmt.Errorf("cannot set value on a map")
 }
 
 func (mi *MapItem) SetValue(val interface{}) error {
-	if isMapOrArrayItem(val) {
-		return fmt.Errorf("cannot set map-or-array-item value (%T) into mapitem", val)
-	}
-	mi.Value = val
-	return nil
+	mi.ResetValue()
+	return mi.AddValue(val)
 }
 
 func (a *Array) SetValue(val interface{}) error {
 	return fmt.Errorf("cannot set value on an array")
 }
 
 func (ai *ArrayItem) SetValue(val interface{}) error {
-	if isMapOrArrayItem(val) {
-		return fmt.Errorf("cannot set map-or-array-item value (%T) into arrayitem", val)
+	ai.ResetValue()
+	return ai.AddValue(val)
+}
+
+func isValidValue(val interface{}) bool {
+	switch val.(type) {
+	case *Map, *orderedmap.Map,
```

Comment:
nit: probably better to use `default` case to return false

Reply:
🤔 

So, `default` is optional in Go...
I don't see any semantic difference between a `default` over a final-line return.

Hmmm...

In what way is using the `default` case over a final return statement better?

Is it ... that this `switch` guarantees it covers _all_ cases ... and so is grammatically "tighter?"

Reply:
> Is it ... that this switch guarantees it covers all cases ... and so is grammatically "tighter?"

this is purely my readability bias. feel free to ignore. in this case seems like we are trying to cover all cases, hence default in my head is natural exit point to have. (if default is added, thankfully go would be smart enough to see that all cases have returns and be happy with that function without trailing return.)

Reply:
Yup, that makes sense. I appreciate turning over even these little things... however subjective or not. Please keep including these kinds of reflections alongside the meatier ones; I enjoy it all.